### PR TITLE
[CORE] Add Teslamate integration

### PR DESCRIPTION
**Pull request by Zegorax**
Ciao,

Hope you're doing well!

You can find in this PR an integration with Teslamate in addition to Teslalogger. It's using the Factory pattern and interface extensions to have a clean code and working regardless of what the backend provider is.

I made a little modification regarding the distance mechanism. I've changed the check and instead of relying on the distance, it will check whether there is a "fixed" lat/long in the shared link DB (The form allows the creation without one). Therefore, if there is nothing, it will use the destination set in the Tesla.

Feel free to review it and don't hesitate to ask me if you have any question :)

And of course, please test it extensively!

### CODE REVIEW

**Comment by flosoft** [comment]
Awesome stuff! Thanks for your work on this!
I'll give it a test with my TeslaLogger later today and if all works well, I'll merge and build.

File: docker-compose.yml
Comment (flosoft) [inline docker-compose.yml:1]: Let's remove this file. We could instead add a docker-compose.yml.teslamate example including TeslaETA.

File: interfaces/backendfactory.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    # static to have a singleton-like
+    provider: IBackendProvider
+
+    def _load_provider(self, name, base_url, car_id):
+        if name == "teslalogger":
+            return teslalogger.TeslaloggerBackendProvider(base_url, car_id)
+        elif name == "teslamate":
+            return teslamate.TeslamateBackendProvider(base_url, car_id)
+        else:
+            raise Exception(f"Unknown backend provider : {name}. Available choices are 'teslalogger' or 'teslamagte'.")
```
Comment (flosoft) [inline interfaces/backendfactory.py:16]: typo in here

**Review by flosoft — CHANGES_REQUESTED** [review]
2 minor changes, but otherwise looks good to go.

File: interfaces/backendfactory.py (hunk 1 above)
Comment (Zegorax) [reply interfaces/backendfactory.py:16]: Good catch with the typo, will fix it.

File: docker-compose.yml
Comment (Zegorax) [reply docker-compose.yml:1]: Indeed, it was used for my development only and got committed by mistake.

**Comment by Zegorax** [comment]
Sorry I just received the notification about it being merged. Thanks!

**Pull request merged**